Mathlib/Order/SuccPred/Basic.lean

```diff
@@ -513,6 +513,10 @@ theorem succ_eq_sInf [CompleteLattice α] [SuccOrder α] (a : α) :
     exact le_top
   · exact sInf_le (lt_succ_iff_ne_top.2 ha)
 
+theorem succ_eq_iInf [CompleteLattice α] [SuccOrder α] (a : α) : succ a = ⨅ (b) (_ : a < b), b := by
```

Comment:
Does this work now?
```suggestion
theorem succ_eq_iInf [CompleteLattice α] [SuccOrder α] (a : α) : succ a = ⨅ b > a, b := by
```

Reply:
It does! That honestly kinda justifies having these lemmas at all for me.